docs/adr/save-and-exit.md

```diff
@@ -0,0 +1,73 @@
+# Save and Exit
+
+- Status: **proposed**
+- Deciders: Jen Duong (jenbutongit), Enda McCormack (endamccormack)
+- Date: 2022-05-22 (last updated)
+
+## Context and Problem Statement
+
+Users can return to the form runner with pre-existing state using this handler: [initialize session](https://github.com/XGovFormBuilder/digital-form-builder/pull/760/files). There is no known way to exit a form journey mid-way through (a form with incomplete answers) and save this state externally for reuse at some future point.
+
+This request proposes the provision of a mechanism to save and exit from a form. As currently available on form completion and submission, this change should provision an outbound hook to save partially completed form state to a third party service, either periodically (e.g after every question submission) or after the user has explicitly requested to 'save and exit'.
+
+A Save and Exit would include sending partially completed form state to a specified output, possibly using the same output object specified in the json configuration.
+
+## Decision Drivers
+
+- When users are completing large journeys, they may want to 'save and exit' their progress and return to their form for completion at some future point. The form builder does not persist state permanently so it would be useful to have the partial state captured in a permanent store for form rehydration at some future point.
+
+## Considered Options
+
+- save and exit option/button on summary page
+- save and exit option/button on each page
+- autosave flag: an automatic hook when there is a question post on each page
+- prompt the user on form exit to 'save and exit'
+
+## Decision Outcome
+
+Chosen option (draft): "autosave", because this provides the most up-to-date state from the runner to an external service on the most regular basis possible.
+
+- need to decide how the autosave flag should be implemented (env var?)
+- need to decide the criteria for the autosave hook (should it be on every question or every section etc)
+
+### Positive Consequences
+
+- An external service will be provided partial form state so they can utilise the rehydration hook
+
+### Negative Consequences
+
+- higher network traffic between form runner and external service
```

Comment:
I think this would have more implications to your external service. You're essentially saving duplicate entries of stuff per page, which you might not need to if a user decides to sit down and do it all at once. 

If you really want to do this, I wonder if you could for your external service listen to events fired by redis and save it yourself to your external DB?